docs/documentation/best-practices.md

```diff
@@ -14,13 +14,72 @@ Only store primitive data types (including strings) in JobDataMap to avoid data
 ### Use the Merged JobDataMap
 
 The JobDataMap that is found on the `JobExecutionContext` during Job execution serves as a convenience.
-It is a merge of the JobDataMap found on the JobDetail and the one found on the Trigger, with the value in the latter overriding any same-named values in the former.
+The data in the JobDataMap is a merger of the JobDetail and the Trigger, with the data in the Trigger overriding
+any same-named value in the Job.
 
 Storing JobDataMap values on a Trigger can be useful in the case where you have a Job that is stored in the scheduler for regular/repeated use by multiple Triggers,
 yet with each independent triggering, you want to supply the Job with different data inputs.
 
-In light of all of the above, we recommend as a best practice the following: Code within the `IJob.Execute(..)` method should generally retrieve
-values from the JobDataMap on found on the JobExecutionContext, rather than directly from the one on the JobDetail.
+We recommend that code within the `IJob.Execute(..)` method should retrieve
+values from the `MergedJobDataMap` on the `JobExecutionContext`, rather than directly 
+from the JobDetail or Trigger.
+
+```csharp
+public class SomeJob : IJob 
+{
+    public static readonly JobKey Key = new JobKey("job-name", "group-name");
+
+    public Task Execute(IJobExecutionContext context) 
+    {
+        // don't do this
+        var badMethod = context.JobDetail.JobDataMap.GetString("a-value");
+        var alsoBadMethod = context.Trigger.JobDataMap.GetString("a-value");
+
+        // do this
+        var goodMethod = context.MergedJobDataMap.GetString("a-value");
+    }
+}
+```
+
+## Job Tips
+
+### Static Job Key
+
+To simplify `JobKey` access we recommend defining a static field that allows
+easy access to the job's key.
+
+```csharp
+public class SomeJob : IJob 
+{
+    public static readonly JobKey Key = new JobKey("job-name", "group-name");
+
+    public Task Execute(IJobExecutionContext context) { /* elided */ }
+}
+```
+
+then later you can trigger the job directly with
+
+```csharp
+public async Task DoSomething(IScheduler schedule, CancellationToken ct)
+{
+    await schedule.TriggerJob(SomeJob.Key, ct)
```

Comment:
this is exactly what I use, especially if I have an API endpoint like "trigger data synchronization" for our development UI, when you have knowledge of job and the configuration phase has been done already, when configuring you can always have the `.ForJob(theJobInstanceFromJobBuilder)`